client/src/components/game/gui/panels/chat/ChatPanel.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useRef, useEffect } from "react";
+import PropTypes from "prop-types";
+import "./ChatPanel.scss";
+import PubSub from "pubsub-js";
+import { ApplicationState, GUIState, PlayerState } from "../../../../../shared/state/States";
+import { PANEL_CHANGE } from "../../../../../shared/EventTypes";
+import Panels from "../PanelsEnum";
+
+function ChatPanel({ onCloseCallback }) {
+    const [chats, sendChat] = useState([]);
+    const chatContentsRef = useRef(null);
+    const chatInputRef = useRef(null);
+
+    useEffect(() => {
+        const subs = [
+            PubSub.subscribe(PANEL_CHANGE, () => {
+                console.log("panel change from chats");
+                if (GUIState.activePanel === Panels.Chat) {
+                    chatInputRef.current.focus();
+                }
+            }),
+        ];
+
+        return () => {
+            subs.forEach((sub) => {
+                PubSub.unsubscribe(sub);
+            });
+        };
+    });
+
+    const scrollChatToBottom = () => {
+        chatContentsRef.current.scrollTop = chatContentsRef.current.scrollHeight;
+    };
+
+    const sendC = (e) => {
+        if (e.key === "Enter") {
+            const text = e.target.value;
+            if (!text) return;
+
+            const newChat = `${PlayerState.displayName}: ${text}`;
+            sendChat((prevChats) => [...prevChats, newChat]);
+
+            ApplicationState.connection.sendEvent("chat", text);
+
+            e.target.value = "";
+
+            // add some delay to properly scroll down to edge of chats
+            setTimeout(scrollChatToBottom, 10);
+        }
```

Comment:
For logging any messages received (all nearby players, including sender), see https://github.com/Simperfy/dungeonz/blob/feature/chat-module/client/src/network/websocket_events/WebSocketEvents.js#L66

Probably need some kind of "CHAT_MESSAGE" event to publish from there and listen for in this component.
That websocket event currently receives a data object of:
`{ id: 105583, message: "some example text" }`

Where the id is the entity id of the player character on the server. Every player character sprite has a `displayName` attached to it that can be used to get the display name by entity id.
`dungeonz.gameScene.dynamics[data.id].spriteContainer.displayName.text` to get the actual display name value.

See https://github.com/Simperfy/dungeonz/blob/feature/chat-module/client/src/network/websocket_events/PlayerValues.js#L47 for this recently used for the character renaming panel.

Reply:
Will look into it. 👀